libs/server/Resp/Objects/SortedSetCommands.cs

```diff
@@ -450,7 +450,83 @@ private unsafe bool SortedSetScore<TGarnetApi>(int count, byte* ptr, ref TGarnet
         }
 
         /// <summary>
-        /// Removes and returns the first element from the sorted set stored at key, 
+        /// Returns the score of member in the sorted set at key.
+        /// If member does not exist in the sorted set, or key does not exist, nil is returned.
+        /// </summary>
+        /// <typeparam name="TGarnetApi"></typeparam>
+        /// <param name="count"></param>
+        /// <param name="ptr"></param>
+        /// <param name="storageApi"></param>
+        /// <returns></returns>
+        private unsafe bool SortedSetScores<TGarnetApi>(int count, byte* ptr, ref TGarnetApi storageApi)
+            where TGarnetApi : IGarnetApi
+        {
+            ptr += 13;
+
+            //validation if minimum args
+            if (count - 2 == 0)
+            {
+                // send error to output
+                return AbortWithWrongNumberOfArguments("ZMSCORE", count);
+            }
+            else
+            {
+                // Get the key for SortedSet
+                if (!RespReadUtils.ReadByteArrayWithLengthHeader(out var key, ref ptr, recvBufferPtr + bytesRead))
+                    return false;
+
+                if (NetworkSingleKeySlotVerify(key, true))
+                {
+                    var bufSpan = new ReadOnlySpan<byte>(recvBufferPtr, bytesRead);
+                    if (!DrainCommands(bufSpan, count)) return true;
+                    return true;
+                }
+
+                // Prepare input
+                var inputPtr = (ObjectInputHeader*)(ptr - sizeof(ObjectInputHeader));
+
+                //save values
+                var save = *inputPtr;
+
+                // Prepare length of header in input buffer
+                var inputLength = (int)(recvBufferPtr + bytesRead - (byte*)inputPtr);
+
+                int inputCount = count - 2;
+                // Prepare header in input buffer
+                inputPtr->header.type = GarnetObjectType.SortedSet;
+                inputPtr->header.SortedSetOp = SortedSetOperation.ZMSCORE;
+                inputPtr->count = inputCount;
+                inputPtr->done = 0;
+
+                // Prepare GarnetObjectStore output
+                var outputFooter = new GarnetObjectStoreOutput { spanByteAndMemory = new SpanByteAndMemory(dcurr, (int)(dend - dcurr)) };
+
+                var status = storageApi.SortedSetScore(key, new ArgSlice((byte*)inputPtr, inputLength), ref outputFooter);
+
+                //restore input
+                *inputPtr = save;
+
+                switch (status)
+                {
+                    case GarnetStatus.OK:
+                        //process output
+                        var objOutputHeader = ProcessOutputWithHeader(outputFooter.spanByteAndMemory);
+                        ptr += objOutputHeader.bytesDone;
+                        break;
+                    case GarnetStatus.NOTFOUND:
```

Comment:
If key is not found, this is the error that should be displayed: "WRONGTYPE Operation against a key holding the wrong kind of value" (You might need to add it to CmdStrings...)

Reply:
For reference I've checked the latest Redis docker image and for missing keys it will return `nil` on `ZSCORE` and a list of `nil` on `ZMSCORE` if the key does not exist.

I'm not sure if response parity is a goal here so I'm just asking about that(the documentation doesn't appear to be explicit about the missing key case).

I've also noticed that Garnet returns `nil` for `ZSCORE` currently even though it properly catches `NOTFOUND` and is supposedly writing the error response...

I can make both calls consistent and address that issue if it's desirable.

Reply:
Ah, so `RESP_ERRNOTFOUND` seems to be an alias for return `nil`. In any case I can update both to be consistent with Redis, the new error message, or etc.

Please confirm.

Reply:
Yes, we do strive to have parity with the existing RESP protocol responses, makes life easier for everyone (and especially when using 3rd party libraries to test Garnet against). 

And it looks like you're right, you should be returning Nil even when the key doesn't exist. Not sure where I got that error message before...

Reply:
I have updated the output to match the observed behavior from Redis; an array of nil matching the input member count.